test/v1/summary.test.js

```diff
@@ -0,0 +1,138 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import neo4j from '../../src/v1';
+import sharedNeo4j from '../internal/shared-neo4j';
+
+describe('result summary', () => {
+
+  let driver, session;
+
+  beforeEach(done => {
+    driver = neo4j.driver('bolt://localhost', sharedNeo4j.authToken);
+    session = driver.session();
+
+    session.run("MATCH (n) DETACH DELETE n").then(done);
+  });
+
+  afterEach(() => {
+    driver.close();
+  });
+
+  it('should get result summary', done => {
+    // When & Then
+    session.run("CREATE (p:Person { Name: 'Test'})").then(result => {
+      var summary = result.summary;
+
+      expect(summary.statement.text).toBe("CREATE (p:Person { Name: 'Test'})");
+      expect(summary.statement.parameters).toEqual({});
+
+      expect(summary.statementType).toBe("w");
+      expect(summary.plan).toBe(false);
+      expect(summary.profile).toBe(false);
+      expect(summary.notifications).toEqual([]);
+      expect(summary.resultConsumedAfter).toBeDefined();
+      expect(summary.resultAvailableAfter).toBeDefined();
+
+      var counters = summary.counters;
+      expect(counters.nodesCreated()).toBe(1);
+      expect(counters.nodesDeleted()).toBe(0);
+      expect(counters.relationshipsCreated()).toBe(0);
+      expect(counters.relationshipsDeleted()).toBe(0);
+      expect(counters.propertiesSet()).toBe(1);
+      expect(counters.labelsAdded()).toBe(1);
+      expect(counters.labelsRemoved()).toBe(0);
+      expect(counters.indexesAdded()).toBe(0);
+      expect(counters.indexesRemoved()).toBe(0);
+      expect(counters.constraintsAdded()).toBe(0);
+      expect(counters.constraintsRemoved()).toBe(0);
+      done();
+    });
+  });
+
+  it('should get plan from summary', done => {
+    session.run("EXPLAIN MATCH (n) RETURN 1").then(result => {
+      var summary = result.summary;
+      expect(summary.plan).toBeDefined();
+      expect(summary.profile).toBe(false);
+
+      var plan = summary.plan;
+      expect(plan.arguments).toBeDefined();
+      expect(plan.children).toBeDefined();
+      expect(plan.identifiers).toBeDefined();
+      expect(plan.operatorType).toBeDefined();
+      done();
+    });
+  });
+
+  it('should get profile from summary', done => {
+    session.run("PROFILE RETURN 1").then(result => {
+      var summary = result.summary;
+      expect(summary.plan).toBeDefined();
+      expect(summary.profile).toBeDefined();
+
+      var profile = summary.profile;
+      var plan = summary.plan;
+
+      verifyPlansAreEqual(profile, plan);
```

Comment:
Can `isEqual` from lodash be used instead of this function?
I.e. `expect(_.isEqual(profile, plan)).toBeTruthy()`

Reply:
No, my bad, I am actually compare profile and plan. The profile has two extra fields, so the isEqual will not work. Change the method name to `verifyProfileAndPlanEqual` instead.